### ISSUE TITLE
provider/aws: AWS DB Instance can't have backup retention period of zero

### ISSUE THREAD

**Issue by catsby** [hashicorp/terraform#1044]
The `BackupRetentionPeriod` parameter for [AWS RDS Instance](http://docs.aws.amazon.com/AmazonRDS/latest/APIReference/API_CreateDBInstance.html) allows a `0` value to disable backup retention. While maybe not smart to use, it should probably still be allowed. 

Currently, we set `BackupRetentionPeriod` in [`db_instance.go`][1] like so:

```go
if attr, ok := d.GetOk("backup_retention_period"); ok {
    opts.BackupRetentionPeriod = attr.(int)
    opts.SetBackupRetentionPeriod = true
}
```

Because of the way [GetOk](https://github.com/hashicorp/terraform/blob/master/helper/schema/resource_data.go#L77) works, it interprets a `0` value as "not there", and set's `ok` here to false... so this parameter does not get set, and AWS defaults it to `1`. 

Regular `Get` won't help, because it returns `0` if `backup_retention_period` isn't present... 

[1]: https://github.com/hashicorp/terraform/blob/master/builtin/providers/aws/resource_aws_db_instance.go

**Comment by phinze** [comment]
We talked about a valid workaround in Slack today:

```go
const backupRetentionPeriodUnset = -1

// ...

"backup_retention_period": &schema.Schema{
  // ...
  Default: backupRetentionPeriodUnset,
}

// ...

brp := d.Get("backup_retention_period")
if brp != backupRetentionPeriodUnset {
  opts.SetBackupRetentionPeriod = true
  opts.BackupRetentionPeriod = brp.(int)
}
```

Doesn't address the general case that we should probably consider whether we need to design a mechanism for differentiating between unset and zero-value. But gets this particular issue unblocked.

**Comment by mitchellh** [comment]
That sounds good to me.

**Comment by mitchellh** [comment]
@catsby Fixed?

**Comment by catsby** [comment]
This is fixed in #1043
